cookbooks/aws-parallelcluster-slurm/templates/default/slurm/head_node/check_login_nodes_status.sh.erb

```diff
@@ -0,0 +1,79 @@
+# If there are any healthy or unhealthy nodes found, it concludes that there are running login nodes.
+#
+# Usage: ./check_if_has_running_login_nodes.sh
+
+set -e
+
+STACK_NAME="<%= @stack_name %>"
+LOGIN_NODES_POOL_NAME="<%= @login_nodes_pool_name %>"
+REGION="<%= @region %>"
+
+# Check if LOGIN_NODES_POOL_NAME is empty
+if [[ -z "${LOGIN_NODES_POOL_NAME}" ]]; then
+    echo "No LoginNodes are specified. Please refer to ParallelCluster official documentation for more information."
+    exit 0
+fi
+
+# List all Load Balancers
+load_balancers=$(aws elbv2 describe-load-balancers --region ${REGION})
+
+# Iterate over Load Balancers to find the one with matching tags
+load_balancer_arn=''
+for arn in $(echo "${load_balancers}" | jq -r '.LoadBalancers[].LoadBalancerArn'); do
+    # Get tags for the current Load Balancer
+    tags=$(aws elbv2 describe-tags --resource-arns "${arn}" --region ${REGION})
+
+    # Check if the tags match the desired stack name and login nodes pool name
+    stack_name_match=$(echo "${tags}" | jq -r --arg key "parallelcluster:cluster-name" --arg value "${STACK_NAME}" '.TagDescriptions[] | select(.Tags[]? | (.Key == $key and .Value == $value))')
+    login_nodes_pool_name_match=$(echo "${tags}" | jq -r --arg key "parallelcluster:login-nodes-pool" --arg value "${LOGIN_NODES_POOL_NAME}" '.TagDescriptions[] | select(.Tags[]? | (.Key == $key and .Value == $value))')
+
+    # If both tags are found, store the ARN and break the loop
+    if [[ -n "${stack_name_match}" && -n "${login_nodes_pool_name_match}" ]]; then
+        load_balancer_arn="${arn}"
+        break
+    fi
+done
+
+# Output result
+if [[ -n "${load_balancer_arn}" ]]; then
+    echo "Load Balancer ARN found: ${load_balancer_arn}"
+else
+    echo "No Load Balancer found for the specified stack and login nodes pool."
+    exit 1
+fi
+
+# Get Target Group ARN associated with the Load Balancer
+target_group_arn=$(aws elbv2 describe-target-groups \
+    --load-balancer-arn $load_balancer_arn \
+    --query "TargetGroups[0].TargetGroupArn" \
+    --output text \
+    --region ${REGION})
+
+# Exit if Target Group is not found
+if [[ -n "${target_group_arn}" ]]; then
+    echo "TargetGroup ARN found: ${target_group_arn}"
+else
+    echo "No Target Group found for the specified Load Balancer."
+    exit 1
+fi
+
+# Get the number of healthy and unhealthy targets
+target_healths=$(aws elbv2 describe-target-health \
+    --target-group-arn $target_group_arn \
+    --region ${REGION})
+
+healthy_count=$(echo $target_healths | jq -r '.TargetHealthDescriptions[] | select(.TargetHealth.State == "healthy") | .Target.Id' | wc -l)
+unhealthy_count=$(echo $target_healths | jq -r '.TargetHealthDescriptions[] | select(.TargetHealth.State != "healthy") | .Target.Id' | wc -l)
+
+# Check if there are running login nodes
+total_nodes=$((healthy_count + unhealthy_count))
+if [[ $total_nodes -gt 0 ]]; then
+    echo "Login nodes are running."
+else
+    echo "Login nodes are stopped."
```

Comment:
Shouldn't we exit 0 here? Actually, you could end the if condition earlier, remove the else and just exit 0 after echoing that login nodes are stopped.